com.unity.formats.fbx/Editor/ConvertToNestedPrefab.cs

```diff
@@ -526,6 +535,56 @@ internal static List<GameObject> GetSceneReferencesToObject(Object obj)
             return prefab;
         }
 
+        internal struct SourceObjectInfo
+        {
+            public GameObject destGO;
+#if UNITY_2021_2_OR_NEWER
+            public List<Object> sceneObjectsWithReference;
+#endif // UNITY_2021_2_OR_NEWER
+
+            public SourceObjectInfo(GameObject dest)
+            {
+                destGO = dest;
+#if UNITY_2021_2_OR_NEWER
+               sceneObjectsWithReference = new List<Object>();
+#endif // UNITY_2021_2_OR_NEWER
+        }
+    }
+
+        internal static Dictionary<string, SourceObjectInfo> s_nameToInfo = new Dictionary<string, SourceObjectInfo>();
+
+        internal static void GatherSceneHierarchy(GameObject hierarchyRoot)
+        {
+            s_nameToInfo.Clear();
+
+#if UNITY_2021_2_OR_NEWER
+            var isPreviewScene = SceneManagement.EditorSceneManager.IsPreviewSceneObject(hierarchyRoot);
+#endif // UNITY_2021_2_OR_NEWER
+
+            var q = new Queue<Transform>();
+            q.Enqueue(hierarchyRoot.transform);
+            while (q.Count > 0)
+            {
+                var t = q.Dequeue();
+
+                var info = new SourceObjectInfo();
+#if UNITY_2021_2_OR_NEWER
+                if (!isPreviewScene)
+                {
+                    var instanceID = t.gameObject.GetInstanceID();
+                    info.sceneObjectsWithReference = GetSceneReferences(instanceID);
+                }
+#endif // UNITY_2021_2_OR_NEWER
+
+                s_nameToInfo[t.name] = info;
```

Comment:
How do you set sceneObjectsWithReference in 2020.3 ?

Reply:
Also: I assume we renamed objects in the hierarchy before we get to this point, right?

Reply:
I don't because in 2020.3 searching for the object does not return results of objects that reference its components

Reply:
OK now I get it. Can you document that somehow, that you can search by object in recent Unity but in older Unity you need to search by component?